src/components/forms/cbr/CbrRuleForm.js

```diff
@@ -0,0 +1,302 @@
+                disableSave: this.props.resourceAttributeProps.disableSave,
+                type: "formInSubForm",
+              }}
+            />
+            {/* tags */}
+            <IcseFormTemplate
+              name="Tags"
+              subHeading
+              tooltip={{
+                content: "A cloud resource can also be specified using tags.",
+                align: "top-left",
+              }}
+              addText="Create a Tag"
+              arrayData={this.props.data.tags}
+              innerForm={CbrTagForm}
+              disableSave={this.props.tagProps.disableSave}
+              onDelete={this.props.tagProps.onDelete}
+              onSave={this.props.tagProps.onSave}
+              onSubmit={this.props.tagProps.onSubmit}
+              propsMatchState={this.props.propsMatchState}
+              innerFormProps={{ ...tagInnerFormProps }}
+              hideAbout
+              toggleFormProps={{
+                hideName: true,
+                submissionFieldName: "tags",
+                disableSave: this.props.tagProps.disableSave,
+                type: "formInSubForm",
+              }}
+            />
+          </>
+        )}
+      </div>
+    );
+  }
+}
+
+CbrRuleForm.defaultProps = {
+  data: {
+    name: "",
+    description: "",
+    enforcement_mode: "Enabled",
+    api_type_id: "",
+    contexts: [],
+    resource_attributes: [],
+    tags: [],
+  },
+};
+
+CbrRuleForm.propTypes = {
+  isModal: PropTypes.bool,
+  data: PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    description: PropTypes.string.isRequired,
+    enforcement_mode: PropTypes.string.isRequired,
+    api_type_id: PropTypes.string.isRequired,
+    contexts: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        value: PropTypes.string.isRequired,
+      }).isRequired
+    ),
+    resource_attributes: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        value: PropTypes.string.isRequired,
+      }).isRequired
+    ),
+    tags: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        operator: PropTypes.string.isRequired,
+        value: PropTypes.string.isRequired,
+      }).isRequired
+    ),
+  }),
+  propsMatchState: PropTypes.func.isRequired,
+  invalidCallback: PropTypes.func.isRequired,
+  invalidTextCallback: PropTypes.func.isRequired,
+  invalidNameCallback: PropTypes.func.isRequired,
+  invalidNameTextCallback: PropTypes.func.isRequired,
+  contextProps: PropTypes.shape({
+    onSave: PropTypes.func.isRequired,
+    onDelete: PropTypes.func.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    disableSave: PropTypes.func.isRequired,
+    invalidCallback: PropTypes.func.isRequired,
+    invalidTextCallback: PropTypes.func.isRequired,
+    invalidNameCallback: PropTypes.func.isRequired,
+    invalidNameTextCallback: PropTypes.func.isRequired,
```

Comment:
not using `invalidContextCallback` any more, instead passing `contextProps.invalidCallback`. i did it this way assuming they'll have different invalid callbacks, since i'm not sure how we'd differentiate fields with the same name in different components in craig. it's probably possible